ark/util/serialize.ts

```diff
@@ -8,7 +8,7 @@ export type SerializationOptions = {
 	onCycle?: (value: object) => string
 	onSymbol?: (value: symbol) => string
 	onFunction?: (value: Function) => string
-	onUndefined?: string
+	onUndefined?: string | undefined
```

Comment:
Is there a reason we have to add  `undefined` here explicitly when the key is optional?

Reply:
`exactOptionalPropertyTypes` is enabled in ark util

https://github.com/arktypeio/arktype/blob/main/ark/util/tsconfig.base.json#L15

<img width="855" alt="Screenshot 2024-09-18 at 7 36 30 PM" src="https://github.com/user-attachments/assets/2a1cefa2-ec5d-439d-96b2-f66eace60cbf">

Reply:
oh wait I can just leave out the default case 🤦

Reply:
fixed